pkg/minikube/assets/addons.go

```diff
@@ -247,10 +248,10 @@ var Addons = map[string]*Addon{
 }
 
 func AddMinikubeDirAssets(assets *[]CopyableFile) error {
-	if err := addMinikubeDirToAssets(constants.MakeMiniPath("addons"), constants.AddonsPath, assets); err != nil {
```

Comment:
I needed some time to think about this PR, and while I like the concept of afero.Fs, I don't think this current approach of embedding it around the minikube code base works for me. I feel that passing nil to a function is generally a sign of a hastily tacked on API. Then again, so is having a MakeMiniPath function in a constants package (not your fault!)

Feel free to correct me - but I don't feel that using the real filesystem here will negatively impact our overall test run time. 

It's less elegant, but what do you think about adding a helper to test_addons.go, which sets up a new filesystem directory for minikube:

```
func setupTestDir() (path, error) {
  path, err := ioutil.TempDir()
  os.Setenv(constants.MinikubeHome, path)
  return path, err
}
```

Then you can gather a list of files written to disk via filepath.Walk, and compare it against expectations:

```
path, err := setupTestDir()
got := filepath.Walk(path)

    if diff := cmp.Diff(got, want); diff != "" {
        t.Errorf("files differ: (-want +got)\n%s", diff)
    }
```

It's certainly less elegant, but what are your thoughts?

Reply:
Thanks for the advice.
Changed the test cases to use a temporary dir for Minikube config as suggested.
It should eliminate the code change in non-relevant files by now.